tuf/pyca_crypto_keys.py

```diff
@@ -0,0 +1,1017 @@
+    # "ValueError/IndexError/TypeError:  When the given key cannot be parsed
+    # (possibly because the passphrase is wrong)."
+    # A passphrase or password is not used to 'private_key', since it should
+    # not be encrypted.
+    try:
+      # 'private_key' (in PEM format) must be converted to a
+      # pyca/cryptography private key object before a signature can be
+      # generated.
+      private_key_object = load_pem_private_key(private_key.encode('utf-8'),
+                                                password=None,
+                                                backend=default_backend())
+   
+      # Calculate the SHA256 hash of 'data' and generate the hash's PKCS1-PSS
+      # signature. 
+      rsa_signer = \
+        private_key_object.signer(padding.PSS(mgf=padding.MGF1(hashes.SHA256()),
+                      salt_length=hashes.SHA256().digest_size), hashes.SHA256())
+                          
+    except ValueError: #pragma: no cover
+      raise tuf.CryptoError('The private key could not be deserialized.')
+ 
+    except cryptography.exceptions.UnsupportedAlgorithm: #pragma: no cover
+      message = 'The private key is encrypted with an unsupported algorithm.'
+      raise tuf.CryptoError(message)
+   
+    # Generate an RSSA-PSS signature.  Raise 'tuf.CryptoError' for the expected
+    # pyca/cryptography exceptions.
+    rsa_signer.update(data)
+    signature = rsa_signer.finalize()
+  
+  else:
+    raise TypeError('The required private key is unset.')
+
+  return signature, method
+
+
+
+
+
+def verify_rsa_signature(signature, signature_method, public_key, data):
+  """
+  <Purpose>
+    Determine whether the corresponding private key of 'public_key' produced
+    'signature'.  verify_signature() will use the public key, signature method,
+    and 'data' to complete the verification.
+    
+    >>> public, private = generate_rsa_public_and_private(2048)
+    >>> data = b'The quick brown fox jumps over the lazy dog'
+    >>> signature, method = create_rsa_signature(private, data)
+    >>> verify_rsa_signature(signature, method, public, data)
+    True
+    >>> verify_rsa_signature(signature, method, public, b'bad_data')
+    False
+
+  <Arguments>
+    signature:
+      An RSASSA PSS signature, as a string.  This is the signature returned
+      by create_rsa_signature(). 
+
+    signature_method:
+      A string that indicates the signature algorithm used to generate
+      'signature'.  'RSASSA-PSS' is currently supported.
+
+    public_key:
+      The RSA public key, a string in PEM format.
+
+    data:
+      Data used by tuf.keys.create_signature() to generate
+      'signature'.  'data' (a string) is needed here to verify 'signature'.
+
+  <Exceptions>
+    tuf.UnknownMethodError.  Raised if the signing method used by
+    'signature' is not one supported by tuf.keys.create_signature().
+    
+    tuf.FormatError. Raised if 'signature', 'signature_method', or 'public_key'
+    are improperly formatted.
+
```

Comment:
NOT Resolved, I think.

PR #311:
 - adds mention of fact that tuf.CryptoError can be raised
 - adds mention of fact that tuf.FormatError can be raised by 'data' being improperly formatted

The obvious exceptions are thus handled in PR #311.

PR #311 does not, however, appear to add any tests that would trigger exceptions, and it is implied in Albert's comment that this is needed. (Is that a matter of coding style? I don't see anything like that in [the style guide](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)...?